### PR TITLE
Resframe 4190 user agent

### PR DESCRIPTION
**Pull request by JoeARO**

### CODE REVIEW

File: lib/mozart_fetcher/http_client.ex [hunk 1]
```diff
@@ -37,7 +37,11 @@ defmodule HTTPClient do
   end
 
   defp make_request(endpoint, headers, options, client) do
-    client.get(endpoint, headers, options)
+    client.get(endpoint, request_headers(headers), options)
+  end
+
+  defp request_headers(headers) do
+    headers ++ [{ 'User-Agent', 'Mozart-Fetcher' }]
```
Comment (samfrench) [inline lib/mozart_fetcher/http_client.ex:44]: Can this be "MozartFetcher" as in the ticket.

Comment (JoeARO) [reply lib/mozart_fetcher/http_client.ex:44]: Apologies will update that now :)

Comment (JoeARO) [reply lib/mozart_fetcher/http_client.ex:44]: Its cause the ticket title says 'Mozart-Fetcher' but you are right

File: test/http_client_test.exs [hunk 2]
```diff
@@ -34,6 +34,11 @@ defmodule HTTPClientTest do
       assert resp.request_url == "http://localhost:8082/foo%20bar?baz=%20true"
     end
 
+    test "the request headers contain a User-Agent header with the value Mozart-Fetcher" do
```
Comment (samfrench) [inline test/http_client_test.exs:37]: Do we need to update the description?

**Review by samfrench — APPROVED** [review]

File: test/http_client_test.exs (hunk 2 above)
Comment (JoeARO) [reply test/http_client_test.exs]: Done thanks

**Review by Raymonoir — APPROVED** [review]

File: test/http_client_test.exs [hunk 3]
```diff
@@ -34,6 +34,11 @@ defmodule HTTPClientTest do
       assert resp.request_url == "http://localhost:8082/foo%20bar?baz=%20true"
     end
 
+    test "the request headers contain a User-Agent header with the value MozartFetcher" do
+      {:ok, resp} = HTTPClient.get("http://localhost:8082/foo/bar")
+      assert Enum.member?(resp.request.headers, {'User-Agent', 'MozartFetcher'}) == true
```
Comment (boonious) [inline test/http_client_test.exs:39]: The following should also be possible, just to beef up your future options : )

```ex
  assert  {'User-Agent', 'MozartFetcher'} in resp.request.headers
```

**Review by boonious — APPROVED** [review]

File: test/http_client_test.exs (hunk 3 above)
Comment (boonious) [reply test/http_client_test.exs:39]: Also, the following is idiomatic (not needing`== true`) as `Enum.member?/2` is a boolean function

```ex
  assert Enum.member?(resp.request.headers, {'User-Agent', 'MozartFetcher'})
```

**Review by Raymonoir — APPROVED** [review]

**Review by ettomatic — APPROVED** [review]

**Pull request merged**